waiter/src/waiter/auth/authentication.clj

```diff
@@ -290,27 +290,30 @@
           {:keys [expires-at]} cookie-metadata
           ;; handle legacy cookies which will not have this value set
           expires-at (or expires-at current-epoch-time)
-          request-host (some-> request utils/request->host utils/authority->host)]
+          request-host (some-> request utils/request->host utils/authority->host)
+          waiter-token? (not-empty (token->token-parameters request-host))]
       (log/info auth-principal "cookie expires at" expires-at "offset is" offset-parsed)
       (cond
         ;; avoid infinite redirect loop
         done
-        (-> {:status http-204-no-content}
-          (utils/attach-waiter-source)
+        (cond-> (utils/attach-waiter-source {:status http-204-no-content})
+          waiter-token?
           (assoc :waiter/token request-host))
```

Comment:
We should not be responding with success for arbitrary host headers.  My preference would be for 404.

Reply:
Added check for invalid token and 404 response.